agents-and-function-calling/bedrock-agents/features-examples/11-get-user-confirmation-before-action-execution/11.3-create-agent-with-API-schema-and-user-confirmation/11.3-create-agent-with-API-schema-and-user-confirmation.ipynb

```diff
@@ -0,0 +1,2187 @@
+   "id": "ada2844c-04fd-4dfc-a87d-0a668afb8f82",
+   "metadata": {},
+   "source": [
+    "## Invoke the Agent for an API that does not have User Confirmation Enabled\n",
+    "\n",
+    "As the first test, we will invoke our agent with a prompt that we know is for an API where user confirmation has not been enabled. Now that we've created the agent, let's use the `bedrock-agent-runtime` client to invoke this agent and perform some tasks."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 201,
+   "id": "fb174c5e-6cbc-4314-8b0b-a51123a0386a",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "## create a random id for session initiator id\n",
+    "session_id:str = str(uuid.uuid1())\n",
+    "enable_trace:bool = False\n",
+    "end_session:bool = False\n",
+    "\n",
+    "simple_agent_invoke(\"what are the open claims?\", agent_id, agent_alias_id, session_id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bb157983",
+   "metadata": {},
+   "source": [
+    "## Invoke the Agent for an API that has User Confirmation Enabled"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 190,
+   "id": "5c6d6869",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'ResponseMetadata': {'HTTPHeaders': {'connection': 'keep-alive',\n",
+      "                                      'content-type': 'application/json',\n",
+      "                                      'date': 'Fri, 27 Sep 2024 17:56:45 GMT',\n",
+      "                                      'transfer-encoding': 'chunked',\n",
+      "                                      'x-amz-bedrock-agent-session-id': 'dc32e864-7cf9-11ef-8bb3-0affee3e2ef7',\n",
+      "                                      'x-amzn-bedrock-agent-content-type': 'application/json',\n",
+      "                                      'x-amzn-requestid': 'c23a5116-2593-4e48-a150-d79fb6f7ef86'},\n",
+      "                      'HTTPStatusCode': 200,\n",
+      "                      'RequestId': 'c23a5116-2593-4e48-a150-d79fb6f7ef86',\n",
+      "                      'RetryAttempts': 0},\n",
+      " 'completion': <botocore.eventstream.EventStream object at 0x7fd34950ac80>,\n",
+      " 'contentType': 'application/json',\n",
+      " 'sessionId': 'dc32e864-7cf9-11ef-8bb3-0affee3e2ef7'}\n"
+     ]
+    }
+   ],
+   "source": [
+    "## create a random id for session initiator id\n",
+    "session_id:str = str(uuid.uuid1())\n",
+    "enable_trace:bool = False\n",
+    "end_session:bool = False\n",
+    "\n",
+    "simple_agent_invoke(\"Please send a reminder to the insured about claim claim-006 only. Please reply indicating whether the reminder was sent or not\", agent_id, agent_alias_id, session_id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "81619dd5",
+   "metadata": {},
+   "source": [
+    "Let's create a method to handle the `returnControl` event and call our lambda locally"
```

Comment:
Same here, we should NOT execute the function to invoke the agent with the function result. Instead we should just pass the confirmation and the agent should handle the execution